fido/common.py

```diff
@@ -2,9 +2,20 @@
 from twisted.web.http_headers import Headers
 
 
+def encode_to_bytes(value):
+    """Twisted requires the method, url, headers to be utf-8 encoded"""
+
+    if isinstance(value, six.text_type):
+        return value.encode('utf-8')
+    return value
```

Comment:
fwiw, there's a better version of this function in `yelp_bytes`, though I'd lean towards *requiring* bytes up front instead of allowing both.  twisted *always* requires bytes